### ISSUE TITLE
[java] UseDiamondOperator false-positive with Lombok val initializer

### ISSUE THREAD

**Issue by xylo** [pmd/pmd#2710]
**Affects PMD Version:** 6.21.0

**Rule:** [UseDiamondOperator](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#usediamondoperator)

**Description:**

False positive when using Lombok's "val" initializer: You cannot omit the `<String>` (if you want a typed array) but PMD complains.

**Code Sample demonstrating the issue:**
```java
import lombok.val;
import java.util.ArrayList;

val foo = new ArrayList<String>(5);
```

**Similar issues:** #1624

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.
